File: standbyme-api/src/models/Usuario.js
const { Model, DataTypes } = require('sequelize');

const bcrypt = require('bcrypt');

class Usuario extends Model {
    static init(sequelize){
        super.init( {
            nomeUsuario: DataTypes.STRING,
            sobrenomeUsuario: DataTypes.STRING,
            email: DataTypes.STRING,
            senha: DataTypes.STRING,
            telefone: DataTypes.STRING
        }, {
            sequelize
        })
        this.addHook('beforeSave', async client => {
            if (client.senha) {
              client.senha = await bcrypt.hash(client.senha, 8);
            }
          });
          return this;
    }
    
}

module.exports = Usuario;

File: standbyme/lib/models/Usuario.dart
import 'package:standbyme_tcc/models/UsuarioBase.dart';

class Usuario extends UsuarioBase {
  String id;

  Usuario.empty();

  Usuario({this.id, primeiroNome, sobrenome, telefone, email, senha})
      : super(
            primeiroNome: primeiroNome,
            sobrenome: sobrenome,
            telefone: telefone,
            email: email,
            senha: senha) {}

  factory Usuario.fromJson(Map<String, dynamic> parsedJson) {
    return Usuario(
        id: parsedJson['id'],
        primeiroNome: parsedJson['nomeUsuario'],
        sobrenome: parsedJson['sobrenomeUsuario'],
        telefone: parsedJson['telefone'],
        email: parsedJson['email'],
        senha: parsedJson['senha']);
  }
}
